plugin/environment.go

```diff
@@ -247,6 +256,52 @@ func (env *Environment) Deactivate(id string) bool {
 	return true
 }
 
+// Deactivates, then Activates, the plugin with the given id.
+func (env *Environment) RestartPlugin(id string) (manifest *model.Manifest, activated bool, reterr error) {
+	env.Deactivate(id)
+	return env.Activate(id)
+}
+
+// Pings a plugin through go-plugin's Ping() method
+func (env *Environment) PingPluginRPC(id string) error {
+	if p, ok := env.activePlugins.Load(id); ok {
+		ap := p.(activePlugin)
+		if ap.supervisor != nil {
+			return ap.supervisor.Ping()
+		}
+	}
+
+	return fmt.Errorf("plugin not running: %v", id)
+}
+
+// Checks if the plugin process's PID exists and can respond to a signal.
+func (env *Environment) CheckPluginProcess(id string) error {
+	if p, ok := env.activePlugins.Load(id); ok {
+		ap := p.(activePlugin)
+		if ap.supervisor != nil {
+			return ap.supervisor.CheckProcess()
+		}
+	}
+
+	return fmt.Errorf("plugin not running: %v", id)
+}
+
+// Checks for plugin error status by using the two above functions, as well as the HealthCheck hook, if implemented.
+func (env *Environment) GetPluginErrorStatus(id string) error {
```

Comment:
Here is where the different kinds of health checks are made. The flow currently goes:

```
if err1 = RPC ping fails:
  if err2 = plugin process not responding:
    return err2
  else
    return err1

if err3 = HealthCheck hook returns error
  return err3
```